### PR TITLE
Support service cache update

### PR DESCRIPTION
**Pull request by Jason-ZW**
## Features(#2677)
Existing cache files in this directory`(/var/lib/rancher/cache/)` will be updated, other files will not be affected.
- [x] `ros service list --update`
- [x] `ros console list --update`
- [x] `ros engine list --update`
- [x] `ros os list --update`

## Test Points
- Whether the `ros service list --update` is effective.
- Whether the `ros console list --update` is effective.
- Whether the `ros engine list --update` is effective.
- Whether the `ros os list --update` is effective.
- Whether the `ros service up` can  work with the updated cache file.
- Whether the `ros console switch` can  work with the updated cache file.
- Whether the `ros engine switch` can  work with the updated cache file.
- Whether the `ros os upgrade` can  work with the updated cache file.
- After the reboot everything went as expected

### CODE REVIEW

File: cmd/control/console.go [hunk 1]
```diff
@@ -127,7 +133,11 @@ func consoleEnable(c *cli.Context) error {
 
 func consoleList(c *cli.Context) error {
 	cfg := config.LoadConfig()
-	consoles := availableConsoles(cfg)
+	isUpdate := false
+	if c.Bool("update") {
+		isUpdate = true
+	}
+	consoles := availableConsoles(cfg, isUpdate)
```
Comment (niusmallnan) [inline cmd/control/console.go]: `consoles := availableConsoles(cfg, c.Bool("update"))` ?

File: cmd/control/engine.go [hunk 2]
```diff
@@ -272,7 +278,11 @@ func engineEnable(c *cli.Context) error {
 
 func engineList(c *cli.Context) error {
 	cfg := config.LoadConfig()
-	engines := availableEngines(cfg)
+	isUpdate := false
+	if c.Bool("update") {
+		isUpdate = true
+	}
+	engines := availableEngines(cfg, isUpdate)
```
Comment (niusmallnan) [inline cmd/control/engine.go]: Same here

File: cmd/control/os.go [hunk 3]
```diff
@@ -131,7 +144,11 @@ func getImages() (*Images, error) {
 }
 
 func osMetaDataGet(c *cli.Context) error {
-	images, err := getImages()
+	isUpdate := false
+	if c.Bool("update") {
+		isUpdate = true
+	}
+	images, err := getImages(isUpdate)
```
Comment (niusmallnan) [inline cmd/control/os.go]: Same here

File: pkg/util/network/cache.go [hunk 4]
```diff
@@ -43,3 +43,11 @@ func cacheAdd(location string, data []byte) {
 	cacheFile := cacheDirectory + locationHash(location)
 	os.Rename(tempFile.Name(), cacheFile)
 }
+
+func cacheRemove(location string) error {
+	cacheFile := cacheDirectory + locationHash(location)
+	if err := os.Remove(cacheFile); err != nil {
```
Comment (niusmallnan) [inline pkg/util/network/cache.go]: `return os.Remove(cacheFile)` ?

File: cmd/control/console.go (hunk 1 above)
Comment (Jason-ZW) [reply cmd/control/console.go]: Already Modified

File: cmd/control/os.go (hunk 3 above)
Comment (Jason-ZW) [reply cmd/control/os.go]: Already Modified

File: pkg/util/network/cache.go (hunk 4 above)
Comment (Jason-ZW) [reply pkg/util/network/cache.go]: Already Modified

File: cmd/control/engine.go (hunk 2 above)
Comment (Jason-ZW) [reply cmd/control/engine.go]: Already Modified

**Pull request merged**